modes/play/objects/mob.js

```diff
@@ -47,33 +56,54 @@ class Mob extends Obj {
     } else if (action === 'right') {
       return this.moveRight();
     } else if (action === 'jump') {
-      this.jump();
+      return this.jump();
     } else if (action === 'run') {
-      this.run();
+      return this.run();
     }
   }
   
+  selectNextFrame() {
+    if (this.game.now > this.status.lastAnimationTimestamp + this.animationRate) {
+      if (this.status.animationFrame === this.spriteBank[this.status.action].length - 1) {
+        this.status.animationFrame = 0;
+        this.status.lastAnimationTimestamp = this.game.now;
+      } else {
+        this.status.lastAnimationTimestamp = this.game.now;
+        this.status.animationFrame++;
+      }
+    }
+  }
+
 // Movement methods
   moveLeft() {
-      this.velocity.x = accelTo(Math.abs(this.velocity.x), this.moveSpeed, this.accel) * -1;
+    this.selectNextFrame();
```

Comment:
Can `this.selectNextFrame()` go in the Physics loop or something instead of in the action?